product-service/src/main/java/com/extremecoder/productservice/exception/GlobalExceptionHandler.java

```diff
@@ -141,23 +141,21 @@ public ResponseEntity<Object> handleBindException(BindException ex) {
         // todo: modify response
         log.error(ex.getMessage());
 
-        Map<String, Object> body = new HashMap<>();
 
         //Get all errors
         List<String> errors = new ArrayList<>();
 
-        for (ObjectError allError : ex.getBindingResult().getAllErrors()) {
-            if (allError instanceof FieldError) {
-                errors.add(((FieldError) allError).getField() + " " + allError.getDefaultMessage());
-            } else if (allError instanceof ObjectError) {
-                errors.add(allError.getDefaultMessage());
+        ex.getBindingResult().getAllErrors().forEach(objectError -> {
```

Comment:
We could optimize like this:
```java
return ex.getBindingResult().getAllErrors().stream().map(error -> objectError instanceof FieldError ? (FieldError) objectError).getField() + " " + objectError.getDefaultMessage() : objectError.getDefaultMessage() ).collect(Collectors.toList());
```
This way we don't have create an extra variable `List<String> errors = new ArrayList<>()`